prime-router/src/test/kotlin/azure/ReportFunctionTests.kt

```diff
@@ -386,4 +437,49 @@ class ReportFunctionTests {
         verify(exactly = 1) { actionHistory.trackActionSenderInfo(any(), any()) }
         assert(resp.status.equals(HttpStatus.BAD_REQUEST))
     }
+
+    @Test
+    fun `test processRequest invalid Option`() {
+        // setup steps
+        val actionHistory = spyk(ActionHistory(TaskAction.receive))
```

Comment:
I think this is a good example that the function should be refactored at some point for better unit testing.  Leave as is for now.